src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/integration/visits/VisitsMigrationIntTest.kt

```diff
@@ -85,28 +90,34 @@ class VisitsMigrationIntTest : IntegrationTestBase() {
           assertThat(it.toDateTime).isEqualTo(LocalDateTime.parse("2020-01-02T23:30:00"))
         }
       )
-      verify(visitsMigrationService).migrateVisitsByPage(
+      verify(visitsMigrationService).divideVisitsByPage(
         check {
           assertThat(it.body.prisonIds).containsExactly("MDI", "BXI")
           assertThat(it.body.visitTypes).containsExactly("SCON", "OFFI")
           assertThat(it.body.fromDateTime).isEqualTo(LocalDateTime.parse("2020-01-01T01:30:00"))
           assertThat(it.body.toDateTime).isEqualTo(LocalDateTime.parse("2020-01-02T23:30:00"))
-          assertThat(LocalDateTime.parse(it.migrationId)).isCloseTo(LocalDateTime.now(), within(10, ChronoUnit.SECONDS))
-          assertThat(it.estimatedCount).isEqualTo(23_045)
+          assertThat(LocalDateTime.parse(it.migrationId)).isNotNull
+          assertThat(it.estimatedCount).isEqualTo(86)
         }
       )
-      verify(visitsMigrationService, times(24)).migrateVisitsForPage(
+      verify(visitsMigrationService, times(9)).migrateVisitsForPage(
         check {
           assertThat(it.body.filter.prisonIds).containsExactly("MDI", "BXI")
           assertThat(it.body.filter.visitTypes).containsExactly("SCON", "OFFI")
           assertThat(it.body.filter.fromDateTime).isEqualTo(LocalDateTime.parse("2020-01-01T01:30:00"))
           assertThat(it.body.filter.toDateTime).isEqualTo(LocalDateTime.parse("2020-01-02T23:30:00"))
-          assertThat(LocalDateTime.parse(it.migrationId)).isCloseTo(LocalDateTime.now(), within(10, ChronoUnit.SECONDS))
-          assertThat(it.estimatedCount).isEqualTo(23_045)
-          assertThat(it.body.pageSize).isEqualTo(1_000)
-          assertThat(it.body.pageNumber).isBetween(0, 23)
+          assertThat(LocalDateTime.parse(it.migrationId)).isNotNull
+          assertThat(it.estimatedCount).isEqualTo(86)
+          assertThat(it.body.pageSize).isEqualTo(10)
+          assertThat(it.body.pageNumber).isBetween(0, 8)
         }
       )
+
+      val context: KArgumentCaptor<MigrationContext<VisitId>> = argumentCaptor()
+      val visitIdsUpTo86 = (1L..86L).map { it }.toTypedArray()
+
+      verify(visitsMigrationService, times(86)).migrateVisit(context.capture())
+      assertThat(context.allValues.map { it.body.visitId }).containsExactly(*visitIdsUpTo86)
     }
```

Comment:
again, this will be replaced with WireMock asserts, for now check each of our 86 visit ids where supplied to service